src/OpenyModulesManager.php

```diff
@@ -38,28 +39,61 @@ public function __construct(EntityTypeManagerInterface $entity_type_manager) {
    *   Config entity type (node_type, block_content_type, paragraphs_type, etc.)
    * @param string $bundle
    *   Entity bundle machine name.
+   * @param string $field
+   *   Content entity field name that contain reference to bundle.
    *
    * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
-   * @throws \Drupal\Core\Entity\EntityStorageException
    */
-  public function removeEntityBundle($content_entity_type, $config_entity_type, $bundle) {
-    // Remove existing data of content entity.
-    $query = $this->entityTypeManager
-      ->getStorage($content_entity_type)
-      ->getQuery('AND')
-      ->condition('type', $bundle);
+  public function removeEntityBundle($content_entity_type, $config_entity_type, $bundle, $field = 'type') {
+    if ($content_entity_type) {
+      // Remove existing data of content entity.
+      $query = $this->entityTypeManager
+        ->getStorage($content_entity_type)
+        ->getQuery('AND')
+        ->condition($field, $bundle);
 
-    $ids = $query->execute();
-    $storage_handler = $this->entityTypeManager->getStorage($content_entity_type);
-    $entities = $storage_handler->loadMultiple($ids);
-    $storage_handler->delete($entities);
+      $ids = $query->execute();
+      $storage_handler = $this->entityTypeManager->getStorage($content_entity_type);
+      $entities = $storage_handler->loadMultiple($ids);
+      try {
+        $storage_handler->delete($entities);
+      }
+      catch (\Exception $e) {
+        watchdog_exception('openy', $e, "Error! Can't delete content related 
+        to '@entity_type' bundle '@bundle', please remove it manually (Entity ID's - @ids).<br>
+        After this delete config entity @config_entity_type bundle '@bundle'.", [
+          '@entity_type' => $content_entity_type,
+          '@config_entity_type' => $config_entity_type,
+          '@bundle' => $bundle,
+          '@ids' => implode(', ', $ids),
+        ], RfcLogLevel::NOTICE);
+        return;
+      }
+      // TODO: Fix minor issue. After paragraph bundle deleting and restoring
+      // we get additional empty paragraph on node edit page if try to add
+      // paragraph of this type (if content of this paragraph type was in
+      // node).
+      // Proposed solution - cleanup entity reference field tables from
```

Comment:
Solution should be - resave all entities where reference is even possible
This is must have before merging this PR, because we could break the site with enormous number of deleted referenced paragraphs.
Also, before removing - would be nice to add a step with a list of entities and where they are used ( for paragraps ) to let content managers check all will be good after removal <- just a simple table with a list of view/edit

Reply:
- `resave all entities where reference is even possible` - Fixed
- `step with a list of entities` - I think we will provide this step in `openy_system` module on openy components page, I have created `TODO` note in this form about proposed feature.